borsh-rs/borsh/src/de/mod.rs

```diff
@@ -109,8 +109,11 @@ impl BorshDeserialize for String {
     #[inline]
     fn deserialize<R: Read>(reader: &mut R) -> Result<Self, Error> {
         let len = u32::deserialize(reader)?;
-        let mut result = vec![0; len as usize];
-        reader.read_exact(&mut result)?;
+        // TODO(16): return capacity allocation when we have the size of the buffer left from the reader.
```

Comment:
`read_exact` is supposed to return `ErrorKind::UnexpectedEof` if there is not enough bytes to fill the buffer. Why this change is needed?